src/lib.rs

```diff
@@ -467,6 +492,13 @@ impl Message {
             _ => Err(Error::InvalidMessage)
         }
     }
+
+    /// Constructs a `Message` by hashing `data` with hash algorithm `H`
+    #[cfg(feature = "bitcoin_hashes")]
+    pub fn from_hashed_data<H: ThirtyTwoByteHash + bitcoin_hashes::Hash>(data: &[u8]) -> Self {
```

Comment:
Hmm how is a method like this used? By explicitly providing the type param? That might be a bit awkward UX?

```rust
let msg = Message::from_hashes_data<sha256::Hash>(&my_data[..]);
```
Or is there another way?

Reply:
Right, that's a bit awkward, but imo still better than having a separate method for each hash function. Given the confusion I should definitely document it better since using a type parameter in such a way is not too common. Alternatively I could just remove it, `Message::from(sha256::Hash::hash(&my_data))` will do the same.

Reply:
Just added an example to the docs. I personally find that, although a bit longer, the first version reads better.

Reply:
LGTM :)